packages/nns/src/token.ts

```diff
@@ -0,0 +1,100 @@
+import { ICPTs } from "../proto/ledger_pb";
+import { E8S_PER_TOKEN, TOKEN_DECIMAL_ACCURACY } from "./constants/constants";
+import { FromStringToTokenError } from "./enums/token.enums";
+
+export const convertStringToE8s = (
+  amount: string
+): bigint | FromStringToTokenError => {
+  // Remove all instances of "," and "'".
+  amount = amount.trim().replace(/[,']/g, "");
+
+  // Verify that the string is of the format 1234.5678
+  const regexMatch = amount.match(/\d*(\.\d*)?/);
+  if (!regexMatch || regexMatch[0] !== amount) {
+    return FromStringToTokenError.InvalidFormat;
+  }
+
+  const [integral, fractional] = amount.split(".");
+
+  let e8s = BigInt(0);
+
+  if (integral) {
+    try {
+      e8s += BigInt(integral) * E8S_PER_TOKEN;
+    } catch {
+      return FromStringToTokenError.InvalidFormat;
+    }
+  }
+
+  if (fractional) {
+    if (fractional.length > 8) {
+      return FromStringToTokenError.FractionalMoreThan8Decimals;
+    }
+    try {
+      e8s += BigInt(fractional.padEnd(8, "0"));
+    } catch {
+      return FromStringToTokenError.InvalidFormat;
+    }
+  }
+
+  return e8s;
+};
+
+export type Token = {
+  symbol: string;
+  name: string;
+};
+
+export const ICPToken: Token = {
+  symbol: "ICP",
+  name: "ICP",
```

Comment:
Mmmh not sure, isn't the name of the ICP token equals to "Internet Computer"?

Reply:
I don't know, we can change that in another PR. We probably need to ask business. At the moment I haven't found anything else apart from ICP that could be related to the token. Hence, I kept both as "ICP".

Reply:
Yeah I bet it's "Internet Computer" because it's the name use on any coin exchange. Did not find the spec though.

Reply:
ok, I'll change

Reply:
done